app/src/test/java/org/simple/clinic/summary/bloodpressures/newbpsummary/NewBloodPressureSummaryViewUpdateTest.kt

```diff
@@ -59,4 +59,33 @@ class NewBloodPressureSummaryViewUpdateTest {
             hasEffects(OpenBloodPressureUpdateSheet(bloodPressureMeasurement) as NewBloodPressureSummaryViewEffect)
         ))
   }
+
+  @Test
+  fun `when see all is clicked, then show blood pressure history screen`() {
+    val bloodPressure1 = PatientMocker.bp(
```

Comment:
This test is deeply coupled to the implementation detail that the number of BPs that will be shown in the list will be 3. In addition, this number is defined as a magic number in the effect handler and is not configurable in any fashion.

This also merges both the fetching of number of BPs and showing the "See All" button to the same flow, which will become hard to maintain and understand going forward.

What I recommend we should do is this:

- Create a `NewBloodPressureSummaryViewConfig` class which will have a single property, say, `numberOfBpsToDisplay`.
- Provide instances of this config object to the `Init`,`Update`, and `ViewRenderer` classes when setting up the test fixtures and mobius delegate.
- Pass in the number of BPs to load as a property of the load BPs effect and use this in the effect handler to fetch the number of BPs.
- Separate out the flow to hide/show the "See All" button by having a separate effect to load the total count of BPs for a given patient. This can then become a new event that is passed to the `Update` class which can then update the model with the total count of BPs.
- This can then be used in the UI renderer to compare with the total number of BPs to display to decide to hide/show the "See All" button.

I understand that we are making two different SQL queries for essentially the same data, but in terms of real world performance impact, this will be less than a couple of hundred ms on most devices. What we gain instead is:

- Clear, separate, flows for both displaying the BPs and toggling the "See All" button
- Discrete tests for both of these flows so that they can be tested independently of each other
- Better separation of both the flows since they are easier to remove/change since they do not affect each other

Reply:
I have couple of questions of about this can we discuss this in person?

Reply:
In this PR `ViewRenderer` is not implemented, It will be implemented in the later PR